### DISCUSSION TITLE
How to use the ComponentCover

### DISCUSSION THREAD

**Discussion by jenniferarnesen** [dhis2/notes#220] [Q&A] [answered]
Hi, I need to block interaction with a component in certain situations. I thought I could do this by wrapping the component in a ComponentCover, but the underlying component is still interactive, and some styles (e.g. button, dashboard item) are showing through the cover (I've applied translucent). I guess the question is - is the above scenario the intended purpose of the Component cover?

![image](https://user-images.githubusercontent.com/6113918/107473709-65e64b80-6b71-11eb-9b86-a5ce1d14fb7b.png)

Could the answer to this question be added to the ui documentation? Thanks!

**Comment by varl** [comment, marked as answer]
Hi @jenniferarnesen,

The usage is to render the `ComponentCover` [inside the component](https://github.com/dhis2/ui/blob/master/packages/core/src/ComponentCover/ComponentCover.stories.js#L25-L32) you want to block off:

```jsx
    <div>
        <ComponentCover />

        <p>Other things in the component that will be covered by the ComponentCover component</p>
    </div>
```

The way it works is that the `ComponentCover` attempts to cover its parent, and not cover its children.

The `ComponentCover` accepts children to provide a way to render things on top of the cover, for example, a spinner. So things that are inside the component cover will get a higher `z-index` than the `ComponentCover` itself.

Does that explain how and why it works the way it does?

**Reply by amcgee** [reply]
This seems like the answer to me, @jenniferarnesen ?

**Reply by jenniferarnesen** [reply]
Yes, thanks that solved it for me.

**Comment by Mohammer5** [comment]
I'm wondering why the `ComponentCover` is being used to prevent user interaction with a certain element. I think the intended use-case of the component cover is to prevent interaction with "every thing else" while allowing interaction with a defined set of components.

If you want to disable the functionality of buttons, etc, wouldn't it be better to add `disabled={true}`, which should also make the component render the correct disabled styles (at least if they're the "new" ui component)?

**Reply by amcgee** [reply]
This is a good question, but tangential to this discussion Q&A I think...

**Reply by amcgee** [reply]
Ah sorry, I misread your comment - I think this is actually a question about @jenniferarnesen's use case specifically not about the design of the ComponentCover component
